onnxruntime/core/providers/xnnpack/math/matmul.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "matmul.h"
+#include "core/providers/cpu/math/matmul_helper.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+
+bool MatMul::IsMatMulOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph) {
+  bool supported = false;
+  const onnxruntime::Node& node = node_unit.GetNode();
+
+  // use do {} while(false) so it's easier to set a breakpoint on the return
+  do {
+    const auto alpha = node.GetAttributes().find("alpha");
+    if ((*alpha).second.f() != 1.0) break;
+
+    auto input_defs = node.InputDefs();
+
+    if (input_defs.size() != 2) {
+      break;
+    }
+
+    const auto& A_arg = *input_defs[0];
+    const auto& B_arg = *input_defs[1];
+
+    // Support only float
+    const auto* A_type = A_arg.TypeAsProto();
+    const auto* B_type = B_arg.TypeAsProto();
+
+    const auto* A_shape = A_arg.Shape();
+    const auto* B_shape = B_arg.Shape();
+
+    if (A_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT ||
+        B_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+      break;
+    }
+
+    if (!A_shape || A_shape->dim_size() != 2 ||
+        (A_shape->dim(0).dim_value() != 1 && A_shape->dim(1).dim_value() != 1)) {
+      break;
+    }
+
+    if (!B_shape || B_shape->dim_size() != 2 || 
+        (B_shape->dim(0).dim_value() != 1 && B_shape->dim(1).dim_value() != 1)) {
+      break;
+    }
+
+    // B matrix must be constant
+    if (!graph.IsConstantInitializer(B_arg.Name(), true)) {
+      break;
+    }
+
+    supported = true;
+
+  } while (false);
+
+  return supported;
+}
+
+MatMul::MatMul(const OpKernelInfo& info) : XnnpackKernel(info) {
+  int64_t temp;
+  ORT_ENFORCE(info.GetAttr<int64_t>("transA", &temp).IsOK());
+  trans_A_ = temp == 0 ? CblasNoTrans : CblasTrans;
+
+  ORT_ENFORCE(info.GetAttr<int64_t>("transB", &temp).IsOK());
+  trans_B_ = temp == 0 ? CblasNoTrans : CblasTrans;
```

Comment:
The MatMul spec doesn't have any attributes for transposing so where are these values coming from? I believe we have an internal fusion with Transpose that can add them, but that's a level 2 optimization that would happen after xnnpack sees the model. 

More concerning is that it's not clear to me how unit tests are passing given this is requiring the attributes to be provided isn't it? Would be good to validate tests are running by deliberating returning an invalid value in the results.